README.md

```diff
@@ -74,6 +79,67 @@ docker run -p 9042:9042  \
   --rm datastax/cql-proxy:v0.1.1 \
   -e ASTRA_TOKEN=<astra-token> -e ASTRA_DATABASE_ID=<astra-datbase-id>
 ```
+
+### Using a configuration file
+
+Proxy settings can also be passed using a configuration file with the `--config /path/to/proxy.yaml` flag. This can be mixed and matched with command-line flags and environment variables. Here are some example configuration files:
+
+```yaml
+contact-points:
+  - 127.0.0.1
+username: cassandra
+password: cassandra
+port: 9042
+bind: 127.0.0.1:9042
+# ...
+```
+
+or with a Astra token:
+
+```yaml
+astra-token: <astra-token>
+astra-database-id: <astra-database-id>
+bind: 127.0.0.1:9042
+# ...
+```
+
+All configuration keys match their command-line flag counterpart, e.g. `--astra-bundle` is
+`astra-bundle`,  `--contact-points` is `contact-points` etc.
+
+#### Setting up peer proxies
+
+Multi-region failover with DC-aware load balancing policy is the most useful case for a multiple proxy setup.
+
+When configuration `peers:` it is required to set `--rpc-address` (or `rpc-address:` in the yaml) for each proxy and it must match is correspond `peers:` entry. Also, `peers:` is only available in the configuration file and cannot be set using a command-line flag.
+
+##### Multi-region setup
+
+Here's an example of configuring multi-region failover with two proxies. A proxy is started for each region in the cluster and connects to it using that region's bundle. They all share a common configuration file that contains the full list of proxies. 
+
+*Note:* Only bundles are supported for multi-region setups.
+
+```sh
+cql-proxy --astra-bundle astra-region1-bundle.zip --username token --passowrd <astra-token> \
+  --bind 127.0.0.1:9042 --rpc-address 127.0.0.1 --data-center dc-1 --config proxy.yaml
+```
+
+```sh
+cql-proxy ---astra-bundle astra-region2-bundle.zip --username token --passowrd <astra-token> \
+  --bind 127.0.0.2:9042 --rpc-address 127.0.0.2 --data-center dc-2 --config proxy.yaml
+```
+
+The peers settings are configured using a yaml file. It's a good idea to explicitly provide the `--data-center` flag, otherwise; these values are pulled from the backend cluster and would need to be pulled from the `system.local` and `system.peers` table to properly setup the peers `data-center:` values. Here's an example `proxy.yaml`:
```

Comment:
I understand that providing this configuration via CLI or ENV variables is very tricky but it would be very helpful to have that even if it is not super user friendly.

In terms of deployment automation it's so much easier to use env variables or cli parameters than files, especially if users start using CLI/ENV and then realize later on that they need to add in a file (at which point it's likely they will want to move everything to the file regardless because mixing configuration formats is a recipe for disaster).

We can always open a ticket so that it might be prioritized later on if users request it.

Reply:
This is a good point. I wonder if I can just make this a json payload (json is a subset of yaml).